default_crypto.go

```diff
@@ -32,48 +32,58 @@ func (c DefaultCrypto) GenerateDH() (DHPair, error) {
 	var pubKey [32]byte
 	curve25519.ScalarBaseMult(&pubKey, &privKey)
 	return dhPair{
-		privateKey: privKey,
-		publicKey:  pubKey,
+		privateKey: privKey[:],
+		publicKey:  pubKey[:],
 	}, nil
 }
 
 // DH returns the output from the Diffie-Hellman calculation between
 // the private key from the DH key pair dhPair and the DH public key dbPub.
-func (c DefaultCrypto) DH(dhPair DHPair, dhPub Key) Key {
+func (c DefaultCrypto) DH(dhPair DHPair, dhPub Key) (Key, error) {
 	var (
 		dhOut   [32]byte
-		privKey [32]byte = dhPair.PrivateKey()
-		pubKey  [32]byte = dhPub
+		privKey [32]byte
+		pubKey  [32]byte
 	)
+	copy(privKey[:], dhPair.PrivateKey()[:32])
+	copy(pubKey[:], dhPub[:32])
+
 	curve25519.ScalarMult(&dhOut, &privKey, &pubKey)
-	return dhOut
+	return dhOut[:], nil
 }
 
 // KdfRK returns a pair (32-byte root key, 32-byte chain key) as the output of applying
 // a KDF keyed by a 32-byte root key rk to a Diffie-Hellman output dhOut.
-func (c DefaultCrypto) KdfRK(rk, dhOut Key) (rootKey, chainKey, headerKey Key) {
+func (c DefaultCrypto) KdfRK(rk, dhOut Key) (Key, Key, Key) {
 	var (
-		r   = hkdf.New(sha256.New, dhOut[:], rk[:], []byte("rsZUpEuXUqqwXBvSy3EcievAh4cMj6QL"))
+		r   = hkdf.New(sha256.New, dhOut, rk, []byte("rsZUpEuXUqqwXBvSy3EcievAh4cMj6QL"))
 		buf = make([]byte, 96)
 	)
 
 	// The only error here is an entropy limit which won't be reached for such a short buffer.
 	_, _ = io.ReadFull(r, buf)
 
+	rootKey := make(Key, 32)
+	var headerKey Key = make(Key, 32)
```

Comment:
I added a comment to one line but here and in the line below we can simplify this like in the line 66.